scripts/upgrade

```diff
@@ -15,123 +12,154 @@ source /usr/share/yunohost/helpers
-  ynh_app_setting_set $app backup_core_only 1
-  
-  ynh_backup_before_upgrade # Backup the current version of the app
-  ynh_clean_setup () {
-      ynh_restore_upgradebackup
-  }
-  ynh_abort_if_errors	# Stop script if an error is detected
+# Normalize the URL path syntax
+path_url=$(ynh_normalize_url_path --path_url=$path_url)
+
+#=================================================
+# STANDARD UPGRADE STEPS
+#=================================================
+# DOWNLOAD, CHECK AND UNPACK SOURCE
+#=================================================
+
+if [ "$upgrade_type" == "UPGRADE_APP" ]
+then
+	ynh_script_progression --message="Upgrading source files..." --weight=3
+
+	# Create tmp directory and fetch app inside
+    tmpdir="$(ynh_smart_mktemp --min_size=300)"
+    ynh_setup_source --dest_dir="$tmpdir"
+
+    # Fetch needed plugins
+    mkdir -p $tmpdir/plugins/Ldap_Login
+    ynh_setup_source --dest_dir="$tmpdir/plugins/Ldap_Login" --source_id=ldap_plugin
+    ynh_setup_source --dest_dir="$tmpdir/plugins" --source_id=log_failed_logins_plugin
 fi
 
 #=================================================
-# INSTALL DEPENDENCIES
+# NGINX CONFIGURATION
 #=================================================
+ynh_script_progression --message="Upgrading nginx web server configuration..." --weight=2
 
-ynh_install_app_dependencies "$pkg_dependencies"
+# Create a dedicated nginx config
+ynh_add_nginx_config
 
 #=================================================
-# DOWNLOAD, CHECK AND UNPACK SOURCE
+# UPGRADE DEPENDENCIES
 #=================================================
+ynh_script_progression --message="Upgrading dependencies..." --weight=6
 
-# Create tmp directory and fetch app inside
-tmpdir=$(ynh_mkdir_tmp)
-ynh_setup_source "$tmpdir"
-# Fetch needed plugins
-mkdir -p $tmpdir/plugins/Ldap_Login
-ynh_setup_source "$tmpdir/plugins/Ldap_Login" ldap_plugin
-ynh_setup_source "$tmpdir/plugins" log_failed_logins_plugin
+ynh_install_app_dependencies "$pkg_dependencies"
 
 #=================================================
 # CREATE DEDICATED USER
 #=================================================
+ynh_script_progression --message="Making sure dedicated system user exists..."
+
+# Create a dedicated user (if not existing)
+ynh_system_user_create --username=$app
 
-ynh_system_user_create $app # Create dedicated user if not existing
+#=================================================
+# PHP-FPM CONFIGURATION
+#=================================================
+ynh_script_progression --message="Upgrading php-fpm configuration..." --weight=3
+
+# Create a dedicated php-fpm config
+ynh_add_fpm_config
 
 #=================================================
-# SPECIFIC SETUP
+# SPECIFIC UPGRADE
 #=================================================
+# COPY FILES TO $FINAL_PATH
+#=================================================
+
+# sets extended pattern matching options in the bash shell
+shopt -s extglob
```

Comment:
That's a bit mysterious what this is used for but okay I guess :P

Reply:
It's explained the line before ;)
It sets the extended pattern matching. That makes bash accept regex.